src/python/pants/scm/git.py

```diff
@@ -567,27 +569,28 @@ def _read_tree(self, path):
       return tree
     tree = {}
     object_type, tree_data = self._read_object_from_repo(rev=self.rev, relpath=path)
-    assert object_type == 'tree'
+    assert object_type == b'tree'
     # The tree data here is (mode ' ' filename \0 20-byte-sha)*
     i = 0
     while i < len(tree_data):
       start = i
-      while tree_data[i] != ' ':
+      while tree_data[i] != b' ':
         i += 1
-      mode = tree_data[start:i]
+      mode = b''.join(tree_data[start:i])
```

Comment:
Originally, mode and the below would be a `List[bytes]`, rather than `bytes`. In Py2, it wouldn't complain about comparing a list to a byte string, but Py3 does.